light-client/src/supervisor.rs

```diff
@@ -10,32 +15,21 @@ use crate::{
     types::{Height, LightBlock, PeerId, Status},
 };
 
-use tendermint::evidence::{ConflictingHeadersEvidence, Evidence};
-
-use contracts::pre;
-use crossbeam_channel as channel;
-
 /// Type alias for readability
 pub type VerificationResult = Result<LightBlock, Error>;
 
-/// Events which are exchanged between the `Supervisor` and its `Handle`s.
+/// Input events sent by the [`Handle`]s to the [`Supervisor`]. They carry a [`Callback`] which is
+/// used to communicate back the responses of the requests.
 #[derive(Debug)]
-pub enum Event {
-    // Inputs
+enum HandleInput {
     /// Terminate the supervisor process
     Terminate(Callback<()>),
     /// Verify to the highest height, call the provided callback with result
     VerifyToHighest(Callback<VerificationResult>),
     /// Verify to the given height, call the provided callback with result
     VerifyToTarget(Height, Callback<VerificationResult>),
-
-    // Outputs
-    /// The supervisor has terminated
-    Terminated,
-    /// The verification has succeded
-    VerificationSuccess(Box<LightBlock>),
-    /// The verification has failed
-    VerificationFailure(Error),
```

Comment:
Interesting. I wasn't fully aware that these are needed for internal communication only. Looks like a good simplification. Especially removing the `pub` from `Events` (now `HandleInput`) makes sense then 👍  I would definitely like to hear what @romac / @brapse think about this, too.

Reply:
Yes they are internal only so a simplification here might makes sense.

Reply:
To give a bit more context, up until now this enum was a mix of events. Some of them used to send callbacked events from the `Handle` up to its `Supervisor`, while others where only used as intermediates inside of `Handle` methods. This led to the use of wildcard match arms in places where only a subset of events where relevant, with undefined or panic behaviour. Additionally it would mask the case where a new variant was added but not accounted for in vital places, e.g. the `Supervisor` run loop. By isolating the enum to the single purpose of `Handle -> Supervisor` comms, all these downsides are removed and we get some help from the compiler when extending the surface between the two of them.

Reply:
Good catch! I wasn't so happy either with having both input and outputs/internal events in the same enum, but didn't see/look for the opportunity to simplify all this. 👍

Reply:
More thoughts on the matter in https://github.com/informalsystems/tendermint-rs/pull/185#pullrequestreview-439830876